test/TestUtilities/Test.Utility/TestDotnetCLiUtility.cs

```diff
@@ -192,36 +195,39 @@ private static void UpdateCliWithLatestNuGetAssemblies(string cliDirectory)
 
         private static void CopyRestoreArtifacts(string artifactsDirectory, string pathToSdkInCli, string configuration)
         {
-            const string restoreProjectName = "NuGet.Build.Tasks";
-            const string restoreTargetsName = "NuGet.targets";
-            const string restoreTargetsExtName = "NuGet.RestoreEx.targets";
+            var fileExtensions = new[] { ".dll", ".pdb", ".targets" };
 
-            var sdkDependencies = new List<string> { restoreProjectName, "NuGet.Versioning", "NuGet.Protocol", "NuGet.ProjectModel", "NuGet.Packaging", "NuGet.LibraryModel", "NuGet.Frameworks", "NuGet.DependencyResolver.Core", "NuGet.Configuration", "NuGet.Common", "NuGet.Commands", "NuGet.CommandLine.XPlat", "NuGet.Credentials", "NuGet.Build.Tasks.Console" };
+            var sdkDependencies = new List<string> { "NuGet.Build.Tasks.Console", "NuGet.CommandLine.XPlat" };
```

Comment:
May I know why are the dependencies reduced into two? Is that because the two assemblies are now ilmerged?
If so, in the real .NET SDK insertion scenario, are the copied NuGet assemblies also get reduced? Thanks!

Reply:
I'm not 100% sure but I'm assuming its because our projects used to be "legacy" C# projects where project references were not transitive.   This meant that if you wanted to collect the outputs of a project and its dependencies, you sometimes needed to get them from each output folder.  However, with SDK-style projects, everything is transitive so the output folder for these two projects contains everything we need.

We are currently inserting three packages/projects into the .NET SDK:

[NuGet.CommandLine.XPlat](https://github.com/dotnet/sdk/blob/main/src/Layout/tool_nuget/tool_nuget.csproj#L10)
[NuGet.Build.Tasks](https://github.com/dotnet/sdk/blob/main/src/Layout/redist/redist.csproj#L26)
[NuGet.Build.Tasks.Console](https://github.com/dotnet/sdk/blob/main/src/Layout/redist/redist.csproj#L27)

Technically `NuGet.Build.Tasks.Console` references `NuGet.Build.Tasks` so the union of the locally built outputs of `NuGet.CommandLine.XPlat` and `NuGet.Build.Tasks.Console` contain what we need to patch an SDK.

This logic does not currently patch the NuGet-based MSBuild project SDK resolver though, perhaps I should add that?  The layout is slightly different so it would need to be special.

[Microsoft.Build.NuGetSdkResolver](https://github.com/dotnet/sdk/blob/main/src/Layout/redist/redist.csproj#L29)

Reply:
> I'm not 100% sure but I'm assuming its because our projects used to be "legacy" C# projects where project references were not transitive. This meant that if you wanted to collect the outputs of a project and its dependencies, you sometimes needed to get them from each output folder. However, with SDK-style projects, everything is transitive so the output folder for these two projects contains everything we need.

Those projects were never legacy, as they were among the first multi targeted projects in production :D 

I think it's just how the copying was implemented, instead of 2 entry points plus all their dependencies, all assemblies were enumerated.

Reply:
Got it~ Thanks for the explanation!